include/h2o.h

```diff
@@ -961,6 +971,13 @@ struct st_h2o_conn_t {
      * callbacks
      */
     const h2o_conn_callbacks_t *callbacks;
+
+#if H2O_USE_CONN_UUID
+    /**
+     * connection UUID (UUIDv4 string representation)
+     */
+    char uuid[37];
+#endif
```

Comment:
I did say elsewhere that this should be a compile-time option. Let me change my position.

I think that UUID should always be supported, but lazily-initialized, so that it would consume CPU cycles only when used. 37 bytes (or possibly 38, if we add a flag to indicate if the UUID has been generated) is nothing compared to TCP send / receive window size.

For regular usage, we can create a function called `h2o_conn_get_uuid` that returns UUID, generating on the fly it is has not yet been generated. In case of probe points, we can explicitly call that function, with a comment stating that it has to be explicitly created as the attached probe reads the memory directly.

WDYT?

Reply:
It sounds much simpler than compile-time option. Added `h2o_conn_get_uuid()`!